### PR TITLE
[Docs] Add Table of Content

### PR DESCRIPTION
**Pull request by AdarshRawat1**
## fixes 
- #4589
# Tasks Done

- [x] Add a table of contents at the top to make navigation a little easier.
> Better Navigation 

## Proposed [Status - Dropped]
- [ ] Describe `p4c` just over the TOC [ Optional ] 
> To provide a quick introduction of what the repository holds. Reason - Directly giving the TOC can be odd.

image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,47 @@
+# p4c: A Comprehensive P4 Compiler
+p4c is a reference compiler for the P4 programming language, supporting both P4-14 and P4-16 versions. It is a crucial component in the P4 ecosystem, enabling developers to compile P4 programs into target-specific configurations for efficient network programming and deployment.
```
Comment (fruffy) [inline README.md:2]: This sounds like LLM prose. Please remove it.

File: README.md [hunk 2]
```diff
@@ -1,10 +1,47 @@
...
 [![Docker Container](https://github.com/p4lang/p4c/actions/workflows/ci-container-image.yml/badge.svg)](https://github.com/p4lang/p4c/actions/workflows/ci-container-image.yml)
 
+
+Table of Contents
+=================
+
+* [p4c](#p4c)
+* [Getting started](#getting-started)
+   * [Installing packaged versions of p4c](#installing-packaged-versions-of-p4c)
+      * [Ubuntu](#ubuntu)
```
Comment (fruffy) [inline README.md:18]: I would not go that deep, one level of nesting is sufficient.

**Review by fruffy — CHANGES_REQUESTED** [review]
Thank you!

File: README.md (hunk 2 above)
Comment (AdarshRawat1) [reply README.md:18]: Sure

File: README.md (hunk 1 above)
Comment (fruffy) [reply README.md:2]: No worries, for now we do not have to add additional descriptions on P4C.

Comment (AdarshRawat1) [reply README.md:2]: Noted sir!

File: README.md (hunk 2 above)
Comment (Abhinavcode13) [reply README.md:18]: https://github.com/p4lang/p4c/tree/main/testdata do add information about testdata too in TOC @AdarshRawat1.

Comment (AdarshRawat1) [reply README.md:18]: @Abhinavcode13 are you referring to  [Test section in README](https://github.com/AdarshRawat1/p4c/tree/Docs-11-Adding-Table-of-content#tests) ??

Context -  I have covered it in TOC, but will remove it now as we are going to cover only 1 level of nesting.

> TOC is supposed to cover only the contents of README, so we won't be including links to directory in TOC.

**Comment by fruffy** [comment]
This will be out of date after https://github.com/p4lang/p4c/pull/4580 so let's wait for that.

**Comment by AdarshRawat1** [comment]
TOC is now in Sync with the updated README structure.

File: README.md [hunk 3]
```diff
@@ -5,6 +5,33 @@
 [![Validation](https://github.com/p4lang/p4c/actions/workflows/ci-validation-nightly.yml/badge.svg)](https://github.com/p4lang/p4c/actions/workflows/ci-validation-nightly.yml)
 [![Docker Container](https://github.com/p4lang/p4c/actions/workflows/ci-container-image.yml/badge.svg)](https://github.com/p4lang/p4c/actions/workflows/ci-container-image.yml)
 
+
+Table of Contents
```
[the remark this answers is not in the document]
Comment (fruffy) [reply README.md:9]: In the sense of 
image
This should say p4c and the p4c in the TOC is removed.

[the remark this answers is not in the document]
Comment (AdarshRawat1) [reply README.md:9]: Sure sir!!

[the remark this answers is not in the document]
Comment (fruffy) [reply README.md]: No need to call me sir. :) 

Also these replies generate a decent amount of email traffic, you do not have to reply to all my comments. Reactions are also okay.

File: README.md [hunk 4]
```diff
@@ -5,6 +5,32 @@
...
+* [Build system](#build-system)
+   * [Defining new CMake targets](#defining-new-cmake-targets)
+* [Known issues](#known-issues)
+   * [Frontend](#frontend)
+   * [Backends](#backends)
+* [How to Contribute](#how-to-contribute)
+* [P4 Compiler Onboarding](#p4-compiler-onboarding)
+* [Contact](#contact)
+
 # p4c
```
Comment (fruffy) [inline README.md:34]: Remove this then we are done. We want p4c at the top and the second header is superfluous. 

image

Comment (AdarshRawat1) [reply README.md]: Sure Sir

> 😅Just finished writing the same question on above review. That what to do with this heading.

**Pull request merged**